src/api/mruby.c

```diff
@@ -0,0 +1,1233 @@
+{
+    mrb_state* mrb = ((mrbVm*)machine->currentVM)->mrb;
+    if (mrb->exc)
+    {
+        mrb_value ex = mrb_obj_value(mrb->exc);
+        mrb_value bt = mrb_exc_backtrace(mrb, ex);
+        if (!mrb_array_p(bt))
+            bt = mrb_get_backtrace(mrb);
+        mrb_value insp = mrb_inspect(mrb, ex);
+        mrb_ary_unshift(mrb, bt, insp);
+        mrb_value msg = mrb_ary_join(mrb, bt, mrb_str_new_cstr(mrb, "\n"));
+        char* cmsg = mrb_str_to_cstr(mrb, msg);
+        machine->data->error(machine->data->data, cmsg);
+        mrb->exc = NULL;
+
+        return false;
+    }
+
+    return true;
+}
+
+static bool initMRuby(tic_mem* tic, const char* code)
+{
+    tic_core* machine = (tic_core*)tic;
+
+    closeMRuby(tic);
+
+    CurrentMachine = machine;
+
+    machine->currentVM = malloc(sizeof(mrbVm));
+    mrbVm *currentVM = (mrbVm*)machine->currentVM;
+
+    mrb_state* mrb = currentVM->mrb = mrb_open();
+    mrbc_context* mrb_cxt = currentVM->mrb_cxt = mrbc_context_new(mrb);
+    mrb_cxt->capture_errors = 1;
+    mrbc_filename(mrb, mrb_cxt, "user code");
+
+#define API_FUNC_DEF(name, _, __, nparam, nrequired, callback, ...) {mrb_ ## name, nrequired, (nparam - nrequired), callback, #name},
+    static const struct{mrb_func_t func; s32 nrequired; s32 noptional; bool block; const char* name;} ApiItems[] = {TIC_API_LIST(API_FUNC_DEF)};
+#undef API_FUNC_DEF
+
+    for (s32 i = 0; i < COUNT_OF(ApiItems); ++i)
+    {
+        mrb_aspec args = MRB_ARGS_NONE();
+        if (ApiItems[i].nrequired > 0)
+            args |= MRB_ARGS_REQ(ApiItems[i].nrequired);
+        if (ApiItems[i].noptional > 0)
+            args |= MRB_ARGS_OPT(ApiItems[i].noptional);
+        if (ApiItems[i].block)
+            args |= MRB_ARGS_BLOCK();
+
+        mrb_define_method(mrb, mrb->kernel_module, ApiItems[i].name, ApiItems[i].func, args);
+    }
+
+    mrb_load_string_cxt(mrb, code, mrb_cxt);
+    return catcherr(machine);
+}
+
+static void evalMRuby(tic_mem* tic, const char* code) {
+    tic_core* machine = (tic_core*)tic;
+
+    mrb_state* mrb = ((mrbVm*)machine->currentVM)->mrb = mrb_open();
+
+    if (!mrb)
+        return;
+
+    if (((mrbVm*)machine->currentVM)->mrb_cxt)
+    {
+        mrbc_context_free(mrb, ((mrbVm*)machine->currentVM)->mrb_cxt);
+    }
+
+    mrbc_context* mrb_cxt = ((mrbVm*)machine->currentVM)->mrb_cxt = mrbc_context_new(mrb);
+    mrb_load_string_cxt(mrb, code, mrb_cxt);
+    catcherr(machine);
+}
+
+static void callMRubyTick(tic_mem* tic)
+{
+    tic_core* machine = (tic_core*)tic;
+    const char* TicFunc = TIC_FN;
+    const char* OvrFunc = OVR_FN;
+
+    mrb_state* mrb = ((mrbVm*)machine->currentVM)->mrb;
+
+    if(mrb)
+    {
+        if (mrb_respond_to(mrb, mrb_top_self(mrb), mrb_intern_cstr(mrb, TicFunc)))
+        {
+            mrb_funcall(mrb, mrb_top_self(mrb), TicFunc, 0);
+            catcherr(machine);
+
+            if (mrb_respond_to(mrb, mrb_top_self(mrb), mrb_intern_cstr(mrb, OvrFunc)))
```

Comment:
We don't need to call `OVR()` here for new languages, it's for compatibility, so we can remove this callback here.